lib/eslint.js

```diff
@@ -413,6 +413,7 @@ function prepareConfig(config) {
         globals: util.mergeConfigs({}, config.globals),
         env: util.mergeConfigs({}, config.env || {}),
         settings: util.mergeConfigs({}, config.settings || {}),
+        disableInlineConfig: config.disableInlineConfig || false,
```

Comment:
I'm not sure adding `disableInlineConfig` to `config` is a good idea. That means people can do that inside of their `.eslintrc` files, and we need to think a lot harder about exposing that.